packages/galeri/src-xpetra/Galeri_Elasticity2DProblem.hpp

```diff
@@ -383,13 +383,13 @@ namespace Galeri {
 
       // Equalize norms of all vectors to that of the first one
       // We do not normalize them as a vector of ones seems nice
-      Teuchos::Array<typename TST::magnitudeType> norms2(numVectors);
+      Teuchos::Array<typename STS::magnitudeType> norms2(numVectors);
+      this->Nullspace_->norm2(norms2);
       for (int i = 1; i < numVectors; i++)
         norms2[i] = norms2[0] / norms2[i];
-      norms2[0] = TST::magnitude(TST::one());
-      this->Nullspace_->norm2(norms2);
+      norms2[0] = STS::magnitude(STS::one());
       Teuchos::Array<SC> norms2scalar(numVectors);
-      for (int i = 1; i < numVectors; i++)
+      for (int i = 0; i < numVectors; i++)
```

Comment:
For the purposes of SA multigrid, this normalization with the first nullspace vector doesn't seem useful to me, as the QR process will handle any poor scaling.

Previously, was the first vector getting scaled by zero?!

Reply:
> For the purposes of SA multigrid, this normalization with the first nullspace vector doesn't seem useful to me, as the QR process will handle any poor scaling.

Sure, it may not be necessary.

> Previously, was the first vector getting scaled by zero?!

Yep, the root of your troubles.